Source/Scene/Cesium3DTileset.js

```diff
@@ -1288,338 +1462,270 @@ define([
-                    }
+    /**
+     * Add all descendants of the starting tile that meet the selection heuristic to selectionState.nextQueue.
+     * Any tiles that have no children, are additive, or meet the tileset's maximum screen space error are added
+     * to selectionState.finalQueue instead
+     */
+    function queueDescendants(tileset, start, frameState, selectionState, outOfCore) {
+        var stack = tempStack;
 
-                    if (!allVisibleChildrenLoaded) {
-                        // TODO : render parent against a plane mask of its visible children so there is no overlap,
-                        //        then the refineToVisible flags can be removed.
-
-                        // Tile does not meet SSE.  Add its commands and push its visible children to the stack.
-                        // The parent tile and child tiles will render simultaneously until all visible children
-                        // are ready. This allows the visible children to stay loaded while other children stream in,
-                        // otherwise if only the parent is selected the subtree may be unloaded from the cache.
-                        selectTile(tileset, t, fullyVisible, frameState);
-
-                        for (k = 0; k < childrenLength; ++k) {
-                            child = children[k];
-                            if (isVisible(child.visibilityPlaneMask)) {
-                                if (child.contentUnloaded) {
-                                    requestContent(tileset, child, outOfCore);
-                                } else {
-                                    stack.push(child);
-                                }
-                            }
+        queueTile(tileset, start, start, frameState, selectionState, stack, outOfCore);
+
+        while (stack.length > 0) {
+            var tile = stack.pop();
+            visitTile(tileset, tile, frameState, outOfCore);
+            queueTile(tileset, start, tile, frameState, selectionState, stack, outOfCore);
+        }
+    }
+
+    /**
+     * Load and add a tile to the proper queue and load and push children to the processing stack
+     */
+    function queueTile(tileset, start, tile, frameState, selectionState, stack, outOfCore) {
+        var nextQueue = selectionState.nextQueue;
+        var finalQueue = selectionState.finalQueue;
+        var maximumScreenSpaceError = tileset._maximumScreenSpaceError;
+        var loadSiblings = tileset.loadSiblings;
+
+        if (tile.hasTilesetContent) {
+            loadTile(tileset, tile);
+            if (!tile.contentReady) {
+                finalQueue.push(tile);
+            } else {
+                var child = tile.children[0];
+                child.visibilityPlaneMask = tile.visibilityPlaneMask;
+                child.updateTransform(tile.computedTransform);
+                child.distanceToCamera = tile.distanceToCamera;
+                stack.push(child);
```

Comment:
While it is more efficient to handle the root of an external tileset upfront like this, can it be removed and just handled with the rest of the traversal? My main worry is that it could be easy to forget which properties to be set on the child.